controllers/profiles/common/object_creators.go

```diff
@@ -218,10 +226,21 @@ func WorkflowPropsConfigMapCreator(workflow *operatorapi.SonataFlow) (client.Obj
 	return workflowproj.CreateNewAppPropsConfigMap(workflow, props), nil
 }
 
-func ConfigurePersistence(serviceContainer *corev1.Container, options *operatorapi.PersistenceOptions, defaultSchema, namespace string) *corev1.Container {
+func ConfigurePersistence(serviceContainer *corev1.Container, config *operatorapi.PersistenceOptions, defaultSchema, namespace string) (*corev1.Container, error) {
+	if config == nil {
+		return serviceContainer, nil
+	}
 	c := serviceContainer.DeepCopy()
-	if options != nil && options.PostgreSql != nil {
-		c.Env = append(c.Env, persistence.ConfigurePostgreSqlEnv(options.PostgreSql, defaultSchema, namespace)...)
+
+	var p *operatorapi.PersistencePostgreSQL
+	if config.PostgreSQL == nil && persistence.WorkflowConfig.GetPostgreSQLConfiguration() == nil {
+		return nil, fmt.Errorf("no persistence specification available in the workflow or in the platform")
+	}
+	if config.PostgreSQL != nil {
+		p = config.PostgreSQL
```

Comment:
If the workflow persistence was set, we need to give the chance to see if a schema was provided, and if it's provided, below we must use that schema instead of the defaultSchema that we receive as parameter. 
In this way we give users the chance to fine tune the schema name for a particular workflow.